component/componenthelper/component.go

```diff
@@ -55,10 +66,25 @@ func (be *baseComponent) Shutdown(ctx context.Context) error {
 	return be.shutdown(ctx)
 }
 
-// NewComponent returns a component.Component that calls the given Start and Shutdown.
-func NewComponent(s *ComponentSettings) component.Component {
+// fromOptions returns the internal settings starting from the default and applying all options.
+func fromOptions(options []Option) *baseSettings {
+	opts := &baseSettings{
+		Start:    func(ctx context.Context, host component.Host) error { return nil },
+		Shutdown: func(ctx context.Context) error { return nil },
+	}
+
+	for _, op := range options {
+		op(opts)
+	}
+
+	return opts
+}
+
+// New returns a component.Component that calls the given Start and Shutdown.
```

Comment:
```suggestion
// New returns a component.Component configured with the provided Options.
```

`Start` and `Shutdown` may or may not be provided here, and more options may be offered in the future.